README.md

```diff
@@ -1,10 +1,58 @@
-# java-ide
-In-browser Java IDE
+# Javabuilder
 
+This is the code that compiles and runs user code from the Java Lab app in code studio.
+Javabuilder is a serverless AWS application with three main parts:
+1. The WebSocket communication layer, handled by AWS API Gateway and AWS SQS
+1. The code execution layer, handled by AWS Lambda
+1. The data layer, handled by Code.org's Dashboard server
 
-# Prerequisites
-You need to install Ruby 2.7.4 to build and/or run the Ruby lambda functions (`javabuilder-authorizer` and `api-gateway-routes`).
-[rbenv](https://github.com/rbenv/rbenv) is a useful tool for installing Ruby 
-and managing multiple versions. Follow the instructions [here](https://github.com/rbenv/rbenv#installing-ruby-versions) to
-use rbenv to install a new Ruby version. You may need to also install [ruby-build](https://github.com/rbenv/ruby-build#readme) to get the latest Ruby
-versions. The `.ruby-version` file sets the local Ruby version for javabuilder to be 2.7.4.
\ No newline at end of file
+For an in-depth description of how these three parts work together, see the
+[Code.org Javabuilder Service Infrastructure](https://docs.google.com/document/d/196aKj947BYZXZH3nGvzHprgWPOoy2Kw1x3sKgziUaSo/edit)
```

Comment:
super nit: can we put a note on the google doc that it's referenced by the readme? So if it gets moved/deleted in the future we know to update here

Reply:
👍